File: android/src/main/kotlin/com/klarna/inapp/sdk/flutter_klarna_inapp_sdk/core/util/KlarnaParamMapper.kt
package com.klarna.inapp.sdk.flutter_klarna_inapp_sdk.core.util

import com.klarna.mobile.sdk.api.KlarnaEnvironment
import com.klarna.mobile.sdk.api.KlarnaRegion
import com.klarna.mobile.sdk.api.KlarnaResourceEndpoint

internal object KlarnaParamMapper {

    fun getEnvironment(param: String?): KlarnaEnvironment? {
        KlarnaEnvironment.values().forEach {
            if (it.name.lowercase() == param?.lowercase()) {
                return it
            }
        }
        return null
    }

    fun getEnvironmentOrDefault(param: String?): KlarnaEnvironment {
        return getEnvironment(param) ?: KlarnaEnvironment.getDefault()
    }

    fun getRegion(param: String?): KlarnaRegion? {
        KlarnaRegion.values().forEach {
            if (it.name.lowercase() == param?.lowercase()) {
                return it
            }
        }
        return null
    }

    fun getRegionOrDefault(param: String?): KlarnaRegion {
        return getRegion(param) ?: KlarnaRegion.getDefault()
    }

    fun getResourceEndpoint(param: String?): KlarnaResourceEndpoint? {
        KlarnaResourceEndpoint.values().forEach {
            if (it.name.lowercase() == param?.lowercase()) {
                return it
            }
        }
        return null
    }

    fun getResourceEndpointOrDefault(param: String?): KlarnaResourceEndpoint? {
        return getResourceEndpoint(param) ?: KlarnaResourceEndpoint.getDefault()
    }
}

File: ios/Classes/Core/Util/KlarnaParamMapper.swift
import KlarnaMobileSDK

internal class KlarnaParamMapper {
    
    static func getEnvironment(param: String?) -> KlarnaEnvironment? {
        if let param = param?.lowercased() {
            return KlarnaEnvironment.value(forKey: param) as? KlarnaEnvironment
        }
        return nil
    }
    
    static func getEnvironmentOrDefault(param: String?) -> KlarnaEnvironment {
        return getEnvironment(param: param) ?? KlarnaEnvironment.production
    }
    
    static func getRegion(param: String?) -> KlarnaRegion? {
        if let param = param?.lowercased() {
            return KlarnaRegion.value(forKey: param) as? KlarnaRegion
        }
        return nil
    }
    
    static func getRegionOrDefault(param: String?) -> KlarnaRegion {
        return getRegion(param: param) ?? KlarnaRegion.eu
    }
    
    static func getResourceEndpoint(param: String?) -> KlarnaResourceEndpoint? {
        if let param = param?.lowercased() {
            switch param {
            case "alternative_1":
                return KlarnaResourceEndpoint.alternative1
            case "alternative_2":
                return KlarnaResourceEndpoint.alternative2
            default:
                return nil
            }
        }
        return nil
    }
    
    static func getResourceEndpointOrDefault(param: String?) -> KlarnaResourceEndpoint {
        return getResourceEndpoint(param: param) ?? KlarnaResourceEndpoint.alternative1
    }
}
